scripts/apply-mail.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python3
+
+import email
+import argparse
+import re
+import tempfile
+import os
+
+argparser = argparse.ArgumentParser()
+argparser.add_argument('emails', metavar='FILE', nargs='+',
+                            help='Files with email message containing a patch')
+
+
+args = argparser.parse_args()
+
+bad_google = re.compile(r'".* via .*" <.*@googlegroups.com>')
+
+temps = []
+
+for email_file in args.emails:
+    e = email.message_from_file(open(email_file))
+    author = e.get('From')
+    m = re.fullmatch(bad_google, author)
+    if m:
+        e.replace_header('From', e.get('Reply-To'))
+    o = tempfile.NamedTemporaryFile()
+    o.write(bytes(e))
+    temps.append(o)
+
+names = str.join(' ', [f.name for f in temps])
+
+os.system(f'git am -3mi {names}')
```

Comment:
Maybe it's better to stop before calling `git am`? So the script is then called `fix-mail-from.py` or something.
I don't always use `-i` key, and sometimes (but maybe it was some other project) I had to use the `--no-scissors` one, so this hard-coded git-am CLI will cause some inconvenience.

Reply:
I think it's good to have the script do the "git am" otherwise I will forget I even need to use it (if I always use "git am" manually, it's easy to forget there's another step before it).

I agree that the "-i" option isn't necessary and I never use it - simply because you can always follow this with "git commit --amend" if you want to edit the commit message. On the other hand, it isn't so hard to exit the editor too, so I think it's fine that "-i" is the default.

Regarding adding **more** options - we could just add all the extra options that we didn't understand to the git am line. I'm not saying this will be easy with the python parseargs library, though :-( Maybe it's easiest just to add specific options (like the "--no-scissors" one you wanted) and copy them to the command line.

Reply:
> On the other hand, it isn't so hard to exit the editor too...

![1_AD1e170YTJaiUBypv1H9Ow](https://user-images.githubusercontent.com/4498177/187708764-a5702a99-270b-47aa-94a6-7b0842a51aaa.jpg)